File: sdk/java-play-framework/app/controllers/ProductLookupApiController.java
package controllers;

import apimodels.InvalidCode;
import apimodels.Item;
import apimodels.NotFound;

import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import play.Configuration;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2020-03-27T18:10:03.884-04:00[America/New_York]")

public class ProductLookupApiController extends Controller {

    private final ProductLookupApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Configuration configuration;

    @Inject
    private ProductLookupApiController(Configuration configuration, ProductLookupApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }


    @ApiAction
    public Result getItemByASIN(String asin) throws Exception {
        Item obj = imp.getItemByASIN(asin);
        if (configuration.getBoolean("useOutputBeanValidation")) {
            OpenAPIUtils.validate(obj);
        }
        JsonNode result = mapper.valueToTree(obj);
        return ok(result);
    }

    @ApiAction
    public Result getItemByEAN(String ean) throws Exception {
        Item obj = imp.getItemByEAN(ean);
        if (configuration.getBoolean("useOutputBeanValidation")) {
            OpenAPIUtils.validate(obj);
        }
        JsonNode result = mapper.valueToTree(obj);
        return ok(result);
    }

    @ApiAction
    public Result getItemByUPC(String upc) throws Exception {
        Item obj = imp.getItemByUPC(upc);
        if (configuration.getBoolean("useOutputBeanValidation")) {
            OpenAPIUtils.validate(obj);
        }
        JsonNode result = mapper.valueToTree(obj);
        return ok(result);
    }
}

File: sdk/scala-play-server/app/api/ProductLookupApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.InvalidCode
import model.Item
import model.NotFound

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2020-03-27T18:11:16.680-04:00[America/New_York]")
@Singleton
class ProductLookupApiController @Inject()(cc: ControllerComponents, api: ProductLookupApi) extends AbstractController(cc) {
  /**
    * GET /magicCashew/barcodable/1.0.0/api/v1/asin/:asin
    * @param asin ASIN code of item to return
    */
  def getItemByASIN(asin: String): Action[AnyContent] = Action { request =>
    def executeApi(): Item = {
      api.getItemByASIN(asin)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /magicCashew/barcodable/1.0.0/api/v1/ean/:ean
    * @param ean EAN code of item to return
    */
  def getItemByEAN(ean: String): Action[AnyContent] = Action { request =>
    def executeApi(): Item = {
      api.getItemByEAN(ean)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /magicCashew/barcodable/1.0.0/api/v1/upc/:upc
    * @param upc UPC code of item to return
    */
  def getItemByUPC(upc: String): Action[AnyContent] = Action { request =>
    def executeApi(): Item = {
      api.getItemByUPC(upc)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}
